forta-bots/L1-Teleport-Backing/src/agent.spec.ts

```diff
@@ -0,0 +1,161 @@
+import { Finding, FindingSeverity, FindingType, HandleTransaction, keccak256 } from "forta-agent";
+import { createAddress, TestTransactionEvent } from "forta-agent-tools/lib/tests";
+import { provideHandleTransaction } from "./agent";
+import { Interface } from "@ethersproject/abi";
+import { MINT_IFACE } from "./utils";
+
+const WRONG_IFACE: Interface = new Interface([
+  "event WrongEvent(address indexed originator, bytes32 indexed hashGUID)",
+]);
+
+const testCreateFinding = (txHash: string, hashGUID: string, networkId: number): Finding => {
+  return Finding.fromObject({
+    name: "MakerDAO Teleport Backing Monitor",
+    description: "Mint event emitted from TeleportJoin without corresponding WormholeInitialized event",
+    alertId: "MK-02-02",
+    protocol: "forta-bots-info",
+    severity: FindingSeverity.High,
+    type: FindingType.Suspicious,
+    metadata: {
+      txHash,
+      hashGUID,
+      chainId: networkId.toString(),
+    },
+  });
+};
+
+describe("WormholeInitialized events monitoring bot test suite", () => {
+  const mockNetworkManager = {
+    TeleportJoin: createAddress("0x0a"),
+    TeleportOracleAuth: createAddress("0x0b"),
+    networkId: 1,
+  };
+  const mockFetcher = {
+    L2HashGUIDExists: jest.fn(),
+  };
+
+  const handleTransaction: HandleTransaction = provideHandleTransaction(mockNetworkManager as any, mockFetcher as any);
+
+  it("should ignore empty transactions", async () => {
+    const txEvent = new TestTransactionEvent();
+
+    const findings: Finding[] = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should ignore other events emitted from TeleportJoin contract", async () => {
+    mockFetcher.L2HashGUIDExists.mockReturnValue(false);
```

Comment:
when using mocks this way is a good practice to check that the function was called with the expected parameters. The syntax is something like:
`expect(mockFetcher.L2HashGUIDExists).calledWith(...)`